### ISSUE TITLE
[router] path matching fails when route parameter includes forward slash

### ISSUE THREAD

**Issue by RoxKilly** [angular/angular#10502]
`Angular 2 rc 4`, using `@angular/router` on `Firefox 48`.  My routes:
```
{path: 'login',  component: LoginComponent},
{path: 'payment/:id',  component: PaymentStatusComponent, canActivate:[AuthGuard], 
{path: '**', component: PageNotFoundComponent}, 
```
If a logged-out user tries to go to `/payment/:id` The `AuthGuard` redirects with:
```
this.router.navigate(['/login',{next:state.url}]); //state.url is the requested url
```
So the user is successfully redirected to `/login;next=/payment/53` and `LoginComponent`. So far so good.

The problem is, if I open a new browser tab and navigate directly to `/login;next=/payment/53`, the router fails to match it against the `LoginComponent`.  Instead, I am taken to the `PageNotFoundComponent` and the browser url bar changes to `/login;next=true/payment/53` 

- Why is the path `/login;next=...` not matched against `LoginComponent` when accessed directly?
- Why is the string `true` prepended to the value of the `next` parameter in this case?

If I replace `;` with `?` and -- in the new browser tab -- navigate to `/login?next=/payment/53`, the router recognizes the route and loads `LoginComponent`

The problem may have to do with the presence of `/` in the parameter. If I try navigating directly to `/login;next=payment53`,  the router successfully matches the path to `LoginComponent`

**Comment by Tragetaschen** [comment]
#8695, #9526

**Comment by vsavkin** [comment]
Fixed here https://github.com/angular/angular/pull/10513

**Comment by kellyrmilligan** [comment]
was this ever fixed for angular 1.6?
